### PR TITLE
Add typings for url-regex-safe

### PR DESCRIPTION
**Pull request by stuartlong**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/url-regex-safe/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    trailingPeriod: boolean;
+    ipv4: boolean;
+    ipv6: boolean;
+    tlds: string[];
+    returnString: boolean;
+}>;
+
+declare function createUrlRegExp(options?: Options): RegExp;
+
+export = createUrlRegExp;
```
Comment (peterblazejewicz) [inline types/url-regex-safe/index.d.ts:22]:
```suggestion
export as namespace urlRegexSafe;

export = createUrlRegExp;
```
I think it requires UMD support:
https://github.com/niftylettuce/url-regex-safe#vanillajs
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library
https://unpkg.com/url-regex-safe@1.0.2/dist/url-regex-safe.min.js

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
@stuartlong looks OK, except of this UMD support, mind amending PR?
Thx!

File: types/url-regex-safe/index.d.ts (hunk 1 above)
Comment (stuartlong) [reply types/url-regex-safe/index.d.ts:24]: Sure thing, thanks!

**Review by peterblazejewicz — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by stuartlong** [comment]
Ready to merge

**Pull request merged**